addons/mail/models/mail_thread.py

```diff
@@ -2067,12 +2067,17 @@ def _notify_record_by_email(self, message, recipients_data, msg_vals=False,
             base_template = False
 
         mail_subject = message.subject or (message.record_name and 'Re: %s' % message.record_name) # in cache, no queries
+        try:
+            parent_id = message.parent_id.message_id if message.parent_id else False
```

Comment:
This isn't actually a parent_id but the message_id of the parent_id. Allowing it to be False looks wrong I think it could break routing of answers. Reading parent_id as sudo looks like a better idea, what do you think @tde-banana-odoo